torch/csrc/jit/passes/lower_tuples.cpp

```diff
@@ -40,6 +48,17 @@ static void VisitNode(Node* n, Node* insert_point) {
     }
     return;
   }
+
+  // undo TupleUnpack(TupleIndex)
+  if (n->kind() == prim::TupleIndex) {
+    auto construct = n->inputs().at(0)->node();
+    JIT_ASSERTM(construct->kind() == prim::TupleConstruct, "tuple index not matched to tuple construct");
```

Comment:
It actually doesn't matter here either way, because this version of the pass (different from RemoveSimpleTuples below), is only run when EnsureNoTuples is called. Meaning that this is going to cause an assert either way.